src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/ProjectJsonBuildIntegratedProjectSystem.cs

```diff
@@ -35,25 +35,40 @@ public class ProjectJsonBuildIntegratedProjectSystem : ProjectJsonBuildIntegrate
             // set project id
             var projectId = VsHierarchyUtility.GetProjectId(envDTEProject);
             InternalMetadata.Add(NuGetProjectMetadataKeys.ProjectId, projectId);
+            InternalMetadata.Add(NuGetProjectMetadataKeys.UniqueName, uniqueName);
+        }
 
-            // Override the JSON TFM value from the csproj for UAP framework
-            if (InternalMetadata.TryGetValue(NuGetProjectMetadataKeys.TargetFramework, out object targetFramework))
+        protected override void UpdateInternalTargetFramework()
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
And when you make it async, you dont need `JTF.Run` since you're executing whole method on UI thread.